src/test/java/uk/nhs/hee/tis/trainee/credentials/service/JwtServiceTest.java

```diff
@@ -0,0 +1,164 @@
+import java.security.SecureRandom;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.Base64;
+import java.util.Date;
+import javax.crypto.SecretKey;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import uk.nhs.hee.tis.trainee.credentials.config.GatewayProperties.IssuingProperties.TokenProperties;
+import uk.nhs.hee.tis.trainee.credentials.dto.CredentialDataDto;
+import uk.nhs.hee.tis.trainee.credentials.dto.ProgrammeMembershipDto;
+import uk.nhs.hee.tis.trainee.credentials.dto.TestCredentialDto;
+
+class JwtServiceTest {
+
+  private static final String PROGRAMME_NAME = "Programme One";
+  private static final LocalDate START_DATE = LocalDate.now().minusYears(1);
+  private static final LocalDate END_DATE = LocalDate.now().plusYears(1);
+
+  private static final String GIVEN_NAME = "Anthony";
+  private static final String FAMILY_NAME = "Gilliam";
+  private static final LocalDate BIRTH_DATE = LocalDate.now();
+
+  private static final String AUDIENCE = "https://test.tis.nhs.uk/audience";
+  private static final String ISSUER = "some-identifier";
+
+  private static final int DEFAULT_CLAIM_COUNT = 5;
+
+  private JwtService service;
+  private JwtParser parser;
+
+  @BeforeEach
+  void setUp() throws NoSuchAlgorithmException {
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.registerModule(new JavaTimeModule());
+    mapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+
+    byte[] keyBytes = new byte[32];
+    SecureRandom.getInstanceStrong().nextBytes(keyBytes);
+    String keyString = Base64.getEncoder().encodeToString(keyBytes);
+    SecretKey key = Keys.hmacShaKeyFor(keyBytes);
+    parser = Jwts.parserBuilder().setSigningKey(key).build();
+
+    TokenProperties properties = new TokenProperties(AUDIENCE, ISSUER, keyString);
+    service = new JwtService(mapper, properties);
+  }
+
+  @Test
+  void shouldGenerateTokenWithDefaultClaims() {
+    record EmptyData() implements CredentialDataDto {
+
+      @Override
+      public Instant getExpiration(Instant issuedAt) {
+        return issuedAt.plus(Duration.ofDays(30));
+      }
+
+      @Override
+      public String getScope() {
+        return "issue.Empty";
+      }
+    }
+
+    final Instant now = Instant.now();
+    String tokenString = service.generateToken(new EmptyData());
+
+    Jwt<?, Claims> token = parser.parse(tokenString);
+    Claims tokenClaims = token.getBody();
+
+    assertThat("Unexpected number of claims.", tokenClaims.size(), is(DEFAULT_CLAIM_COUNT));
+    assertThat("Unexpected token aud.", tokenClaims.getAudience(), is(AUDIENCE));
+    assertThat("Unexpected token iss.", tokenClaims.getIssuer(), is(ISSUER));
+
+    Date issuedAt = tokenClaims.getIssuedAt();
+    long issuedAtDiff = Duration.between(issuedAt.toInstant(), now).toMinutes();
+    assertThat("Unexpected difference between now and iat.", issuedAtDiff, is(0L));
```

Comment:
I was thinking 'that's a bit cheeky' - but then I saw the `.toMinutes()` cast, so fair enough 😁

Reply:
Honestly some of the most "how do I even test this?" Tests I've written for a while.

I've mocked the underlying clock in the past, but we can't do that with plain mockito so I had to make do with this.